From 1146007738f59a3bf6f6495566f6d9459107c6ed Mon Sep 17 00:00:00 2001
From: Lin Jinhan <troy.lin@rock-chips.com>
Date: Tue, 19 Jan 2021 11:10:46 +0800
Subject: [PATCH] crypto: rockchip: ahash: fix statesize error

API export&import is using feedback tfm, statesize should be equal to
feedback tfm's statesize, otherwise it will trigger WARN_ON in testmgr
module.

Change-Id: I450b74549dc962b93a0654d83869366868b78363
Signed-off-by: Lin Jinhan <troy.lin@rock-chips.com>
---
 drivers/crypto/rockchip/rk_crypto_v1_ahash.c | 2 ++
 drivers/crypto/rockchip/rk_crypto_v2_ahash.c | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/drivers/crypto/rockchip/rk_crypto_v1_ahash.c b/drivers/crypto/rockchip/rk_crypto_v1_ahash.c
index db92cac84762..0bc93fde535c 100644
--- a/drivers/crypto/rockchip/rk_crypto_v1_ahash.c
+++ b/drivers/crypto/rockchip/rk_crypto_v1_ahash.c
@@ -321,6 +321,8 @@ static int rk_cra_hash_init(struct crypto_tfm *tfm)
 				 sizeof(struct rk_ahash_rctx) +
 				 crypto_ahash_reqsize(ctx->fallback_tfm));
 
+	algt->alg.hash.halg.statesize = crypto_ahash_statesize(ctx->fallback_tfm);
+
 	return 0;
 }
 
diff --git a/drivers/crypto/rockchip/rk_crypto_v2_ahash.c b/drivers/crypto/rockchip/rk_crypto_v2_ahash.c
index 5149c5443865..ba51341b3821 100644
--- a/drivers/crypto/rockchip/rk_crypto_v2_ahash.c
+++ b/drivers/crypto/rockchip/rk_crypto_v2_ahash.c
@@ -605,6 +605,8 @@ static int rk_cra_hash_init(struct crypto_tfm *tfm)
 				 sizeof(struct rk_ahash_rctx) +
 				 crypto_ahash_reqsize(ctx->fallback_tfm));
 
+	algt->alg.hash.halg.statesize = crypto_ahash_statesize(ctx->fallback_tfm);
+
 	return 0;
 }
 
-- 
2.35.3

